src/app/pages/component-sidenav/component-sidenav.service.ts

```diff
@@ -0,0 +1,13 @@
+import {Injectable} from '@angular/core';
+import {MatSidenav} from '@angular/material/sidenav';
+
+@Injectable({providedIn : 'root'})
+export class ComponentSidenavService {
```

Comment:
What is the responsibility of this class? 
* Should have a class description comment
* Should not use "Service" in the name since it doesn't communicate anything about what it does; I always want us to use class names that describe the responsibility
(e.g., Angular Material doesn't have "RadioService", but instead "UniqueSelectionDispatcher")

Reply:
This service allows for the `ComponentSidenav` and the `ComponentNav` to share access to the `MatSidenav` instance when it is open (it needs to be re-set each time the "over" sidenav is opened). This allows the `ComponentNav` to close the "over" sidenav on mobile when a nav item selection is made.

I guess something like `ComponentSidenavCoordinator` would desribe its responsibility?

Reply:
Could the `ComponentSidenav` grab the `MatSidenav` with a static `ViewChild` so that the `ComponentNav` could inject its parent and access it that way without adding another class?

Reply:
I'll give it a shot.

Reply:
Removed the whole service since it is not needed when using anchors and `routerLink` to navigate.